File: frontend/src/lib/api/sync.ts
import api from "./client";

export interface SyncLog {
  id: string;
  household_id: string;
  provider: string;
  status: string;
  accounts_synced: number;
  transactions_imported: number;
  error_message: string | null;
  started_at: string;
  completed_at: string | null;
}

export interface SyncStatus {
  last_sync: SyncLog | null;
  is_stale: boolean;
  syncing: boolean;
}

export const syncApi = {
  status: () => api.get<SyncStatus>("/sync/status").then((r) => r.data),
  trigger: () => api.post<SyncLog>("/sync/trigger").then((r) => r.data),
  history: () => api.get<SyncLog[]>("/sync/history").then((r) => r.data),
};

File: backend/app/schemas/sync.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class SyncLogResponse(BaseModel):
    id: str
    household_id: str
    provider: str
    status: str
    accounts_synced: int
    transactions_imported: int
    error_message: Optional[str]
    started_at: datetime
    completed_at: Optional[datetime]

    model_config = {"from_attributes": True}


class SyncStatusResponse(BaseModel):
    last_sync: Optional[SyncLogResponse]
    is_stale: bool
    syncing: bool = False
